### PR TITLE
add a --all flag for start, and a run command

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
also broke 'update next' out into separate functions. didnt make any changes, just wanted it a little cleaner.

File: main.go [hunk 1]
```diff
@@ -174,13 +175,44 @@ type UpdateInfo struct {
...
+
+	return aggr, nil
+}
+
+func enumerateAllChildPackages(pkg *gx.Package, pkgs map[string]struct{}) error {
+	return pkg.ForEachDep(func(dep *gx.Dependency, pkg *gx.Package) error {
+		// TODO: subtle bug here where we could miss some packages if they appear as
+		// dependencies in one version of a package, but not in another version with the
+		// same name. probably can worry about this later with a 'normalize' command
+		if _, ok := pkgs[dep.Name]; ok {
```
Comment (Stebalien) [inline main.go:200]: What if this stored a map of hash -> name?

File: main.go [hunk 2]
```diff
@@ -189,11 +221,19 @@ var updateStartCmd = cli.Command{
 			return err
 		}
 
-		if len(c.Args()) == 0 {
+		if len(c.Args()) == 0 && !c.Bool("all") {
```
Comment (Stebalien) [inline main.go:230]: Nit: Really, we want this to be an xor.

File: main.go (hunk 1 above)
Comment (Stebalien) [reply main.go:200]: Actually, that's still going to fail. If an _updated_ package depends on a new package, we'll miss that (and won't update it).

File: main.go [hunk 3]
```diff
@@ -876,6 +925,27 @@ func gitPush(remote string, branch string, dir string) error {
...
+			Name: "no-prompt",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		for {
+			if err := updateNext(c); err != nil {
+				return err
+			}
+			if !c.Bool("no-prompt") {
+				fmt.Println("Press any key to continue...")
```
Comment (Stebalien) [inline main.go]: Press "enter" to continue.

**Review by Stebalien — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (whyrusleeping) [reply main.go:200]: yeah... i think the complexity of the fix is higher than its worth. idk

File: main.go (hunk 2 above)
Comment (whyrusleeping) [reply main.go:230]: oh fine

**Pull request merged**